### ISSUE TITLE
Gargoyle inserts line break before inline images

### ISSUE THREAD

**Issue by angstsmurf** [garglk/garglk#338]
As illustrated by typing IMAGE in the official image unit test at https://www.eblong.com/zarf/glulx/imagetest.gblorb, Gargoyle will always draw inline images on a new line, instead of on the same line as intended, which makes. See the attached comparison screenshots from Gargoyle and Lectrote.

As far as I can tell, there is no support in the Gargoyle code for multiple inline images on the same line, or for the imagealign_InlineUp, imagealign_InlineDown or imagealign_InlineCenter flags.

Lectrote (correct):
![lectrote](https://user-images.githubusercontent.com/5066755/47431769-fd56e300-d79c-11e8-8cd4-9a0301897627.png)

Gargoyle (incorrect):
<img width="767" alt="gargoyle" src="https://user-images.githubusercontent.com/5066755/47431791-0778e180-d79d-11e8-90a2-705c2cd42f5d.png">

**Comment by Tynach** [comment]
I'm not so sure Gargoyle *can* implement this, not the way it's architected right now. For example, lines of text are a fixed number of pixels high, configured in the garglk.ini (or .garglkrc) configuration file.

On the other hand, trying the 'left' and 'right' commands in that test program *almost* works correctly, so maybe it *is* possible? Here's an example of 'almost correct':
![Screenshot_20210614_193535](https://user-images.githubusercontent.com/224197/121984525-f9aa6d00-cd47-11eb-9eab-9648b05af3b2.png)

And in Quixe, from eblong.com:
![Screenshot_20210614_193917](https://user-images.githubusercontent.com/224197/121984680-3ece9f00-cd48-11eb-9ef4-0ea2a516343a.png)

**Comment by angstsmurf** [comment]
Yes, this is obviously more of a feature request than an easy bug fix. And few games use these features.

Having a fixed text height is a pretty substantial limitation, though, which means you can't mix different font sizes, can't have variable paragraph spacing, and so on. Things like the larger headline "ImageTest" in the Lectrote screenshot above can't be done in Gargoyle.

**Comment by Tynach** [comment]
@angstsmurf, I don't see anything in the GLK specification that allows for multiple font sizes anyway. The larger headline in the Lectrote screenshot is just a coincidence; it internally uses Quixe, which has an implementation detail that gives a slightly larger font size when printing headers. I don't think it's something you can control in code though, except by sometimes using headers, sometimes using subheaders, and sometimes using regular text. Would still only give you 3 or so different sizes at most though, and you wouldn't be able to make it non-bold or selectively apply other formatting properties, and you'd be at the mercy of the GLK implementation to use the sizes you want.

**Comment by angstsmurf** [comment]
The GLK spec has the concept of [style hints](https://www.eblong.com/zarf/glk/Glk-Spec-075.html#stream_style_hints) by which games can suggest things like text color, justification or indeed font size (loosely defined as a visible increase or decrease from a standard size.)

Using the eleven GLK styles, this would make it possible for a game to have eleven different font sizes in a single paragraph (or word!).

I haven't looked into this, but I think Gargoyle supports all of the GLK style hints except font size.

**Comment by Tynach** [comment]
@angstsmurf, interesting! I see it also has a standard for 'light' fonts in addition to bold. Support for these things would be neat to see in Gargoyle.

Do you know of any games or test programs that use the font size style hints?

**Comment by angstsmurf** [comment]
Not that I know of, but considering how many games are out there, it would be surprising if there isn't at least one.

This is getting a bit off-topic, and I see there already is a an issue about missing styles here: #201.

I'll just whip up a quick test and attach it.

[StyleTest.gblorb.zip](https://github.com/garglk/garglk/files/6672531/StyleTest.gblorb.zip)
<img width="1188" alt="Skärmavbild 2021-06-17 kl  20 33 38" src="https://user-images.githubusercontent.com/5066755/122455757-07146100-cfad-11eb-9258-a010109f9888.png">

**Comment by curiousdannii** [comment]
Larger fonts are used for titles and I think room names in Inform 7 by default.
